qemu/tests/qemu_guest_agent.py

```diff
@@ -1026,6 +1027,124 @@ def gagent_check_with_fsfreeze(self, test, params, env):
         cmd_del_file = "%s %s" % (params["cmd_del"], tmp_file)
         session.cmd(cmd_del_file)
 
+    @error_context.context_aware
+    def gagent_check_guest_exec(self, test, params, env):
+        """
+        Execute a command in the guest via guest-exec cmd,
+        and check status of this process.
+
+        Steps:
+        1) Change guest-exec related cmd to white list,linux guest only.
+        2) Execute guest cmd and get the output.
+        3) Check the cmd's result and output from return.
+        4) Execute guest cmd and no need to get the output.
+        5) Check the cmd's result from return.
+        6) Issue an invalid guest cmd.
+        7) Check the return result.
+        8) Execute guest cmd with wrong args.
+        9) Check the return result.
+
+        :param test: kvm test object
+        :param params: Dictionary with the test parameters
+        """
+
+        def _guest_cmd_run(guest_cmd, cmd_args=None, env=None,
+                           input=None, capture_output=None):
```

Comment:
I don't think you need params `env` and `input`, this function only servers `gagent_check_guest_exec()`, no where use them currently.